### PR TITLE
Improve TICKscript Editing

### PR DESCRIPTION
**Pull request by alexpaxton**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #2673 
Connect #2645 

### The Problem
- Fixed heights cause log message contents to spill out and overlap
- No enforcement of horizontal scrolling on code mirror editor - logs can be pushed off the screen
- Logs design breaks on small screens

### The Solution
- Removed infinite scroll component from logs
- Limit rendering to the most recent 200 log messages
- Log messages can be any height
- If a key/value list exceeds `300px` height (about 17 items) it becomes scrollable rather than causing the message to be exceptionally tall

### Preview
screen shot 2018-02-05 at 2 12 30 pm

Horizontal Scrolling
screen shot 2018-02-05 at 3 11 58 pm

Smaller Screens
screen shot 2018-02-05 at 2 25 49 pm

### CODE REVIEW

**Comment by russorat** [comment]
Screenshots from a measurement with lots of fields...

image

image

File: ui/src/kapacitor/components/LogsTable.js [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
+      {`${numLogsToRender} Most Recent Logs`}
     </div>
+    <FancyScrollbar
+      autoHide={false}
+      className="logs-table--container fancy-scroll--kapacitor"
+    >
+      {logs
+        .slice(0, numLogsToRender)
+        .map((log, i) =>
+          <LogsTableRow key={log.key} logItem={log} index={i} />
```
Comment (ebb-tide) [inline ui/src/kapacitor/components/LogsTable.js]: won't need `i` or `index` (as per comment on `/LogsTableRow.js`

File: ui/src/kapacitor/components/LogItemKapacitorPoint.js [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-      {objKey}: <span>{objValues[i]}</span>
+  return (
+    <div className="logs-table--column">
+      <h1>
+        {`${objKeys.length} ${name}`}
+      </h1>
+      <div className="logs-table--scrollbox">
+        {objKeys.map((objKey, i) =>
+          <div key={i} className="logs-table--key-value">
+            {objKey}: <span>{objValues[i]}</span>
```
[the remark this answers is not in the document]
Comment (alexpaxton) [reply ui/src/kapacitor/components/LogItemKapacitorPoint.js]: Simplified this one

File: ui/src/kapacitor/components/LogsTableRow.js [hunk 3]
```diff
@@ -10,29 +10,29 @@ import LogItemInfluxDBDebug from 'src/kapacitor/components/LogItemInfluxDBDebug'
 
 const LogsTableRow = ({logItem, index}) => {
   if (logItem.service === 'sessions') {
-    return <LogItemSession logItem={logItem} key={index} />
+    return <LogItemSession logItem={logItem} logIndex={index} />
```
[the remark this answers is not in the document]
Comment (alexpaxton) [reply ui/src/kapacitor/components/LogsTableRow.js]: Chopped these indices out

File: ui/src/kapacitor/components/LogsTable.js (hunk 1 above)
Comment (alexpaxton) [reply ui/src/kapacitor/components/LogsTable.js]: Removed this index as well

**Pull request merged**